File: IntelligentRecruitment_microservice/application-service/src/main/java/com/fedicode/applicationservice/Dto/AdminApplicationStatsResponse.java
package com.fedicode.applicationservice.Dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AdminApplicationStatsResponse {
    private long totalApplications;
    private long pendingApplications;
    private long analyzedApplications;
    private long acceptedApplications;
    private long refusedApplications;
    private long acceptedThisMonth;
    private long refusedThisMonth;
    private double averageScore;
    private Map<String,Long> monthlyData;
}

File: IntelligentRecruitment-UI/src/app/models/Dto/Application/AdminApplicationStatsResponse.ts

export interface AdminApplicationStatsResponse {
  totalApplications: number;
  pendingApplications: number;
  analyzedApplications: number;
  acceptedApplications: number;
  refusedApplications: number;
  acceptedThisMonth: number;
  refusedThisMonth: number;
  averageScore: number;
  monthlyData: { [month: string]: number };
}
